src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -176,20 +175,27 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            DateTime now = DateTime.Now;
-            await InvokeJobFunction(now, false);
-
             // restart the timer with the next schedule occurrence
-            now = DateTime.Now;
-            DateTime nextOccurrence = _schedule.GetNextOccurrence(now);
-            TimeSpan nextInterval = nextOccurrence - now;
+            await InvokeJobFunction(DateTime.Now, false);
+            TimeSpan nextInterval = ScheduleStatus.Next - DateTime.Now;
             StartTimer(nextInterval);
         }
 
-        internal async Task InvokeJobFunction(DateTime lastOccurrence, bool isPastDue = false)
+        /// <summary>
+        /// Invokes the job function.
+        /// </summary>
+        /// <param name="invocationTime">The time of the invocation, likely DateTime.Now.</param>
+        /// <param name="isPastDue">True if the invocation is because the invocation is due to a past due timer.</param>
+        /// <param name="runOnStartup">True if the invocation is because the timer is configured to run on startup.</param>
+        internal async Task InvokeJobFunction(DateTime invocationTime, bool isPastDue = false, bool runOnStartup = false)
         {
             CancellationToken token = _cancellationTokenSource.Token;
-            TimerInfo timerInfo = new TimerInfo(_schedule, _scheduleStatus, isPastDue);
+            ScheduleStatus timerInfoStatus = null;
+            if (ScheduleMonitor != null)
+            {
+                timerInfoStatus = ScheduleStatus;
+            }
```

Comment:
Previously, we'd only maintain the ScheduleStatus if we were storing it in a ScheduleMonitor. But it's also handy even if it's just kept in-memory (it's how I prevent duplicate 'Next' values now).

The API for TimerInfo says that you will not get a ScheduleStatus if your interval is short -- but my change meant you'd get a ScheduleStatus no matter what, so I do this to keep it backwards-compatible.

Is there any reason that I need to do this? I can remove this 'back-compat' piece here if there's no objections.